src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java

```diff
@@ -140,4 +142,65 @@ public abstract void doFilter (HttpExchange exchange, Chain chain)
      */
     public abstract String description ();
 
+    /**
+     * Returns a pre-processing Filter with the given description and
+     * implementation.
+     *
+     * <p>The {@link Consumer filterImpl} is the effective implementation of
+     * the returned Filter and is executed for each {@code HttpExchange} before
+     * the exchange is handled further.
+     *
```

Comment:
Please expand the description to include similar: "before invoking the next filter in the chain, or the exchange handler (if this is the final filter in the chain)".   Similar comment for afterResponse.

Maybe rename the Consumer do something more general ("consumer") - as the number of references to the term "filter" make it a little hard to follow.